application/src/main/java/uk/nhs/hee/tis/revalidation/service/RecommendationTisStatusUpdateService.java

```diff
@@ -0,0 +1,79 @@
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.revalidation.service;
+
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome.APPROVED;
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome.REJECTED;
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationStatus.COMPLETED;
+
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.nhs.hee.tis.revalidation.dto.RecommendationStatusCheckDto;
+import uk.nhs.hee.tis.revalidation.entity.DoctorsForDB;
+import uk.nhs.hee.tis.revalidation.entity.Recommendation;
+import uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome;
+import uk.nhs.hee.tis.revalidation.repository.DoctorsForDBRepository;
+import uk.nhs.hee.tis.revalidation.repository.RecommendationRepository;
+
+@Slf4j
+@Service
+public class RecommendationTisStatusUpdateService {
+
+  @Autowired
+  private RecommendationRepository recommendationRepository;
+  @Autowired
+  private DoctorsForDBRepository doctorsForDBRepository;
+
+  public void updateRecommendationAndTisStatus(
+      final RecommendationStatusCheckDto recommendationStatusCheckDto) {
+
+    String gmcReferenceNumber = recommendationStatusCheckDto.getGmcReferenceNumber();
+    Optional<DoctorsForDB> optionalDoctorsForDB = doctorsForDBRepository
+        .findById(gmcReferenceNumber);
+    Optional<Recommendation> optionalRecommendation = recommendationRepository
+        .findById(recommendationStatusCheckDto.getRecommendationId());
+
+    RecommendationGmcOutcome recommendationGmcOutcome = recommendationStatusCheckDto.getOutcome();
+
+    if (optionalRecommendation.isPresent() && optionalRecommendation.get().getGmcNumber()
+        .equals(gmcReferenceNumber)) {
+      final var recommendation = optionalRecommendation.get();
+
+      //if gmc outcome is approved/rejected, update the relevant recommendation to approved/rejected, tis status to complete
+      if (recommendationGmcOutcome.name().equals(APPROVED.name()) || recommendationGmcOutcome.name()
+          .equals(REJECTED.name())) {
+        recommendation.setOutcome(recommendationGmcOutcome);
+        recommendation.setRecommendationStatus(COMPLETED);
+      }
+
+      //also update the tis status in doctorsfordb to complete
+      if (optionalDoctorsForDB.isPresent()) {
+        final var doctorsForDB = optionalDoctorsForDB.get();
+        doctorsForDB.setDoctorStatus(COMPLETED);
+        doctorsForDBRepository.save(doctorsForDB);
```

Comment:
Before we update the doctorsForDB, we may need to make sure the completed recommendation is the latest one of that doctor. As each doctor may have more than one recommendation.

Reply:
> 
Each doctorsForDB , i.e. each doctor is made unique by their gmc Id, and it is also ensured by @Id annotation of Mongo. So in this collection (i.e. DoctorsFroDB) with a given gmc id, only one record of a particular doctor is possible. Multiple recommendations are there in the "Recommendation" collection which I am taking care of in the code  @DorisWongNHS